cmd/commands/upgrade.go

```diff
@@ -0,0 +1,160 @@
+	var releaseInfo ReleaseInfo
+
+	resp, err := http.Get(apiURL)
+	if err != nil {
+		return releaseInfo, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return releaseInfo, err
+	}
+
+	err = json.Unmarshal(body, &releaseInfo)
+	return releaseInfo, err
+}
+
+func constructBinaryName(tagName string) string {
+	osType := strings.ToLower(runtime.GOOS)
+	if osType == "darwin" {
+		osType = "macos"
+	}
+
+	archType := runtime.GOARCH
+	if archType == "arm64" {
+		archType = "aarch64"
+	}
+
+	fileName := fmt.Sprintf("vfox_%s_%s_%s.tar.gz", tagName[1:], osType, archType)
+	return fileName
+}
+
+// constructDiffURL constructs a GitHub diff URL between two version tags
+func constructDiffURL(oldTag, newTag string) string {
+	return fmt.Sprintf("https://github.com/version-fox/vfox/compare/%s...%s", oldTag, newTag)
+}
+
+func getUrls(apiURL string, currVersion string) (string, string) {
+	version := currVersion
+	releaseInfo, err := getLatestReleaseInfo(apiURL)
+	if err != nil {
+		fmt.Println("Error fetching release info:", err)
+		return "", ""
+	}
+	fileName := constructBinaryName(releaseInfo.TagName)
+	binURL := fmt.Sprintf("https://github.com/version-fox/vfox/releases/download/%s/%s", releaseInfo.TagName, fileName)
+	diffURL := constructDiffURL(version, releaseInfo.TagName)
+	return binURL, diffURL
+}
+
+func downloadFile(filepath string, url string) error {
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	_, err = io.Copy(out, resp.Body)
+	return err
+}
+
+func extractTarGz(gzipFilePath, destDir string) error {
+	decompressor := util.NewDecompressor(gzipFilePath)
+	if err := decompressor.Decompress(destDir); err != nil {
+		return err
+	} else {
+		return nil
+	}
+}
+
+func upgradeCmd(ctx *cli.Context) error {
+	apiURL := "https://api.github.com/repos/version-fox/vfox/releases/latest"
+	releaseInfo, err := getLatestReleaseInfo(apiURL)
+	currVersion := fmt.Sprintf("v%s", ctx.App.Version)
+	latestVersion := releaseInfo.TagName
+	fmt.Println("Current version: ", currVersion)
+	fmt.Println("Latest available:", latestVersion)
+
+	if currVersion == latestVersion {
+		return cli.Exit("vfox is already at latest version.", 0)
+	}
+	exePath, err := os.Executable()
+	if err != nil {
+		return cli.Exit("Failed to get executable path: "+err.Error(), 1)
+	}
+
+	binURL, diffURL := getUrls(apiURL, currVersion)
+	exeDir := filepath.Dir(exePath)
+	tempFile := "latest_vfox.tar.gz"
+	tempDir := filepath.Join(exeDir, "update_vfox")
+
+	if err := downloadFile(tempFile, binURL); err != nil {
+		return cli.Exit("Failed to download file: "+err.Error(), 1)
+	}
+	if err := extractTarGz(tempFile, tempDir); err != nil {
+		return cli.Exit("Failed to extract file: "+err.Error(), 1)
+	}
+	newExePath := filepath.Join(tempDir, "vfox")
+	if err := os.Rename(newExePath, exePath); err != nil {
```

Comment:
If you directly replace it, it should be possible under `unix-like` system, but is it also possible under `windows`? Will it not prompt that the current file(`exePath`) is occupied? The current process has not ended yet.